internal/jennies/terraform/templates/types/models.tmpl

```diff
@@ -0,0 +1,14 @@
+package provider
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+{{ $package := .Schema.Package | formatObjectName }}
+{{ range .Objects | }}
+type {{ formatModelName $package .Name }}DataSourceModel struct {
```

Comment:
This kind of renaming should be done with a compiler pass

Reply:
Or not be done, because the disambiguation is handled by putting things into their own packages.